scanner/src/main/kotlin/Scanner.kt

```diff
@@ -79,6 +84,18 @@ class Scanner(
         require(scannerWrappers.isNotEmpty() && scannerWrappers.any { it.value.isNotEmpty() }) {
             "At least one ScannerWrapper must be provided."
         }
+
+        scannerWrappers.values.flatten().distinct().forEach { scannerWrapper ->
+            scannerWrapper.criteria?.run {
+                val rangesList = createRangesList(minVersion, maxVersion)
+                val version = Semver.parse(scannerWrapper.details.version) ?: return@forEach
+
+                require(rangesList.isSatisfiedBy(version)) {
+                    "The version range [$minVersion, $maxVersion) specified in criteria for scanner " +
+                            "'${scannerWrapper.details.name}' is not satisfied by the available version: $version"
+                }
```

Comment:
As discussed at the OCD, I'm not sure whether this approach of failing hard is a good one, or whether we instead simply shouldn't try to *write* scan results outside of `[$minVersion, $maxVersion)` to the scan storage and emit a warning about that. That way locally still a newer scanner version could be tested while having a scan storage configured. And there would be some nice symmetry in that both read and write operations adhere to `[$minVersion, $maxVersion)`. I'm curious about the feedback from @mnonnenmacher here.

Reply:
I think it's better to fail if the configured criteria don't match the scanner details, because it could be very difficult for the user to figure out what's going wrong. For testing a new version it's not so much effort to adjust the version range in the config. And if you don't want to add the results to the storage it's also easy to adjust the list of storage writers.

Reply:
@mnonnenmacher this was my reasoning as well. The cost of misconfiguration is IMHO just to high to tolerate it.
Look at what I did: 
- I filed a bug for this `ConstraintViolation` log entries.  
- I stopped my task of running my scan of some project until I have time to investigate the issue
- did some debugging and found the cause after a while

If I had a misconfigured CI I would have cause potentially many hours of unneeded execution time due to the cache being basically disable unintendedly.